src/system_info/system_info_instance.cc

```diff
@@ -67,9 +69,11 @@ void SystemInfoInstance::InstancesMapInitialize() {
   RegisterClass<SysInfoDeviceOrientation>();
   RegisterClass<SysInfoDisplay>();
   RegisterClass<SysInfoLocale>();
-  RegisterClass<SysInfoNetwork>();
   RegisterClass<SysInfoPeripheral>();
-#if defined(TIZEN_IVI) || defined(TIZEN_MOBILE)
+#ifdef GENERIC_DESKTOP
+  RegisterClass<SysInfoNetworkDesktop>();
+#else
+  RegisterClass<SysInfoNetworkTizen>();
```

Comment:
There should not be two different classes SysInfoNetworkTizen and SysInfoNetworkDesktop, all should be SysInfoNetwork. We just use different suffix(_desktop, _tizen, _ivi, _mobile) to let gyp to choose different files to compile. 

This is common practice in chromium world, you could find lots of places does the same thing, for eg: `src/net/net_util*`